### DISCUSSION TITLE
Matter HVAC virtual device

### DISCUSSION THREAD

**Discussion by macjl** [arendst/Tasmota#22033] [Support request]
Hello,

Would it be possible to add HVAC to Matter virtual devices ?
My plan is then to add a script, that would send IR commands, to control an old AC through Matter.

**Comment by sfromis** [comment]
Does Matter even support such a device type?

**Reply by lboue** [reply]
- [Air Purifier](https://github.com/project-chip/connectedhomeip/tree/master/examples/air-purifier-app/linux)
The Air Purifier example demonstrates a fully functional Matter Air Purifier which is a composed device with: 

  1. Endpoint 1 being the Air Purifier/FanControl
  2. Endpoint 2 is an Air Quality Sensor
  3. Endpoint 3 is a Relative Humidity Sensor
  4. Endpoint 4 is a Temperature Sensor
  5. Endpoint 5 is a Thermostat

- [Thermostat](https://github.com/project-chip/connectedhomeip/tree/master/examples/thermostat/linux)

So you have to deal with some these clusters.


**Matter certified HVAC**
Some HVAC appliances are already Matter certified:
- [Certified Room Air Conditioner](https://csa-iot.org/csa-iot_products/?p_keywords=&p_type%5B%5D=17&p_type%5B%5D=14&p_type%5B%5D=1053&p_device_type%5B%5D=1219&p_program_type%5B%5D=1049&p_certificate=&p_family=)


- [TCL Window AC](https://csa-iot.org/csa_product/tcl-window-ac-20/)
  ![image](https://github.com/user-attachments/assets/4a3315d9-4320-4baa-bb44-4fef88c901d0)

**Comment by s-hadinger** [comment]
Currently it's not supported, but it would be a nice addition indeed, to have Matter -> IRHvac interface.

I'm currently working on the Matter -> Zigbee interface, this will come afterwards
